taqueria-vscode-extension-web-ui/src/data-editors/DataEditorNode.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { isObject } from '../Helpers';
+import { BoolEditor } from './BoolEditor';
+import { ListEditor } from './ListEditor';
+import { MapEditor } from './MapEditor';
+import { getFriendlyDataType } from './MichelineEditor';
+import { OptionEditor } from './OptionEditor';
+import { PairEditor } from './PairEditor';
+import { PrimitiveEditor } from './PrimitiveEditor';
+
+export const DataEditorNode = (
+	{ dataType, value, onChange, hideDataType }: {
+		dataType: any;
+		value: any;
+		onChange: (value: any) => void;
+		hideDataType?: boolean | undefined;
+	},
+) => (
+	<table>
+		<tr>
+			{hideDataType !== true && (
+				<td>
+					<h4 style={{ display: 'inline' }}>{getFriendlyDataType(dataType)}</h4>
+				</td>
+			)}
+			<td>
+				{getEditor({ dataType, onChange, value })}
+			</td>
+		</tr>
+	</table>
+);
+
+const getEditor = (
```

Comment:
It's a bad practice to call functions returnig jsx. It should a be a component that you call in a `<JSX />` fashion with passing props, so that React can propely cache and keep in DOM existing elements on rerender.

Reply:
https://stackoverflow.com/questions/46592833/how-to-use-switch-statement-inside-a-react-component

Reply:
https://dev.to/igor_bykov/react-calling-functional-components-as-functions-1d3l